packages/smooth_app/lib/generic_lib/widgets/smooth_text_form_field.dart

```diff
@@ -74,18 +74,19 @@ class _SmoothTextFormFieldState extends State<SmoothTextFormField> {
       onChanged: widget.onChanged ??
           (String data) {
             // Rebuilds for changing the eye icon
-            if (widget.type == TextFieldTypes.PASSWORD) {
-              if (data.isEmpty) {
-                setState(() {});
-              } else if (data.isNotEmpty && data.length > 1) {
-                setState(() {});
-              }
+            if (widget.type == TextFieldTypes.PASSWORD &&
+                (data.isEmpty || (data.isNotEmpty && data.length > 1))) {
```

Comment:
```suggestion
                data.length != 1) {
```

I don't know the history behind that but we could probably just test if it's a PASSWORD field regardless of the data size.

Reply:
I don't know either, and that's why I left it as it was (logic-wise)

Reply:
> I don't know either, and that's why I left it as it was (logic-wise)

Then please apply my "suggested change": 100% compatible with the previous code but we'll save time for everyone reading this line in the future.